### PR TITLE
Add initial version of the neo4j module

### PR DESCRIPTION
**Pull request by duboisph**
Initial neo4j setup. For now limited to setting up a cluster containing core nodes.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  count       = "${var.core_count}"
+  device_name = "/dev/xvdf"
+  volume_id   = "${aws_ebs_volume.core.*.id[count.index]}"
+  instance_id = "${module.core.instance_ids[count.index]}"
+}
+
+module "puppet_userdata" {
+  source              = "github.com/skyscrapers/terraform-skyscrapers//puppet-userdata?ref=1.0.1"
+  amount_of_instances = "${var.core_count}"
+  customer            = "${var.customer == "" ? "" : "${var.customer}-"}${var.project}"
```
Comment (SamClinckspoor) [inline main.tf]: I think you have a syntax error here

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    content_type = "text/x-shellscript"
+
+    content = <<EOF
+#!/bin/bash
+
+# Wait for the EBS volumes to become ready
+aws --region $(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone | sed 's/[a-z]$//') ec2 wait volume-in-use --filters Name=attachment.instance-id,Values=$(curl -s http://169.254.169.254/latest/meta-data/instance-id) Name=attachment.device,Values=/dev/xvdf
+
+# Create FS
+mkfs.ext4 /dev/xvdf -L NEO4J
```
Comment (SamClinckspoor) [inline main.tf]: this will be ok if an instance is recreated with an existing ebs volume containing data?

**Comment by SamClinckspoor** [comment]
is it possible to set `var.core_amount` to 0? I don't think we need all clusters from prod also in staging/

**Review by SamClinckspoor — APPROVED** [review]

File: sg.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
+resource "aws_security_group" "sg" {
+  name        = "${var.project}-${var.environment}-${var.name}"
```
Comment (iuriaranda) [inline sg.tf]: can you use `name_prefix`? That way you make sure you don't have name collisions and it's more future-proof imo

Comment (venturel) [reply sg.tf]: Do we really like/want to use `name_prefix`?

Comment (SamClinckspoor) [reply sg.tf]: I prefer prefix as well. its always unique and not too long

Comment (venturel) [reply sg.tf]: ok fair enough. My only concern was about how meaningful the name will become but I guess that with tags that's fixed

Comment (duboisph) [reply sg.tf]: Done

Comment (iuriaranda) [reply sg.tf]: yes indeed, we normally have the tags to identify and differentiate resources, and we should keep names as generic as possible imo

**Review by iuriaranda — APPROVED** [review]

**Pull request merged**